### DISCUSSION TITLE
Error when using a custom motif annotation in addMotifAnnotations()

### DISCUSSION THREAD

**Discussion by xmliu01** [GreenleafLab/ArchR#1932] [Questions / Documentation] [answered]
Hello, I am using ArchR to analysis my single-cell ATAC-seq data in wheat, but we I using the addMotifAnnotations(), I got the bellow errors.

ArchR logging to : ArchRLogs/ArchR-addMotifAnnotations-64d158ef102d-Date-2023-04-20_Time-21-49-47.log
If there is an issue, please report to github with logFile!
2023-04-20 21:49:57 : Gettting Motif Set, Species : , 0.183 mins elapsed.
2023-04-20 21:49:59 : Finding Motif Positions with motifmatchr!, 0.207 mins elapsed.
2023-04-20 22:17:30 : All Motifs Overlap at least 1 peak!, 27.726 mins elapsed.
2023-04-20 22:17:30 : Creating Motif Overlap Matrix, 27.726 mins elapsed.
**Error in Matrix::sparseMatrix(i = queryHits(overlapMotifs), j = match(names(allPositions),  : 
  'i' and 'j' must not contain NA**

Because there isn't a motifSet for wheat, so I provided a custom motif list using motifPWMs. 
I provided a custom motif set in meme format and used the read_meme() in package universalmotif to read it.
And then I converted it into PWMatrix using convert_motifs() in package TFBSTools.
Lastly, I used PWMatrixList() in TFBSTools package to convert PWMatrix into PWMatrixList.
So, I thought the format of this custom motif set is correct.
But when I run addMotifAnnotations(), I go the above error.

And here is my code:

`library(universalmotif)`
`library(TFBSTools)`
`library(chromVAR)`
`library(motifmatchr)`
`motif=read_meme("~/TF-motif/meme/Ath_TF_binding_motifs.meme")`
`motif_all=convert_motifs(motif,class="TFBSTools-PWMatrix")`
`tair.motif=do.call(PWMatrixList,motif_all)`
`projSpike2=addMotifAnnotations(projSpike2,name="Motif",motifPWMs=tair.motif)`

So, if anyone knows how to solve this problem.

And I also attached my log file.
[ArchR-addMotifAnnotations-64d158ef102d-Date-2023-04-20_Time-21-49-47.log](https://github.com/GreenleafLab/ArchR/files/11295349/ArchR-addMotifAnnotations-64d158ef102d-Date-2023-04-20_Time-21-49-47.log)

**Comment by xmliu01** [comment, marked as answer]
I thought I have solved this problem.
I found that the names of tair.motif is NULL, and it will got the NA results.
So, if I add the names to tair.motif using names(tair.motif)=ID(tair.motif), then it worked.

The complete code is
```library(universalmotif)
library(TFBSTools)
library(chromVAR)
library(motifmatchr)
motif=read_meme("~/TF-motif/meme/Ath_TF_binding_motifs.meme")
motif_all=convert_motifs(motif,class="TFBSTools-PWMatrix")
tair.motif=do.call(PWMatrixList,motif_all)
names(tair.motif)=ID(tair.motif)
projSpike2=addMotifAnnotations(projSpike2,name="Motif",motifPWMs=tair.motif)
```

And you can also download the motifSet from JASPAR using TFBSTools
```
library(TFBSTools)
library(JASPAR2020)
tair.motif=getMatrixSet(x=JASPAR2020,opts=list(collection="CORE",species="3702",matrixtype="PWM"))
projArchR=addMotifAnnotations(projArchR,name="Motif",motifPWMs=tair.motif,force=T)
```

**Comment by Spirometra** [comment]
Hiya, I came across this potential solution when trying to add the JASPAR2022 set as motifs.
I used:
tair.motif=getMatrixSet(x=JASPAR2022,opts=list(collection="CORE",species="9606",matrixtype="PWM"))
Only get PWMatrixList of length 692 for human.

Regardless, I think it would be great to add JASPAR2022 to the addMotifAnnotations options.
